cpp/daal/include/data_management/data/internal/numeric_table_sycl_homogen.h

```diff
@@ -346,24 +352,30 @@ class DAAL_EXPORT SyclHomogenNumericTable : public SyclNumericTable
                 return status;
             }
 
-            // TODO: Figure out how to convert the data without fallback to host
-            auto hostPtr = buffer.toHost(data_management::readOnly);
+            auto hostPtr = buffer.toHost(data_management::readOnly, status);
+            services::throwIfPossible(status);
+            DAAL_CHECK_STATUS_VAR(status);
+
             internal::VectorUpCast<U, T>()(nRows * nCols, hostPtr.get(), block.getBlockPtr());
 
-            return services::Status();
+            return status;
         }
 
         static services::Status write(services::internal::Buffer<U> buffer, const BlockDescriptor<T> & block, size_t nRows, size_t nCols)
         {
+            services::Status status;
+
             DAAL_ASSERT(block.getNumberOfRows() == nRows);
             DAAL_ASSERT(block.getNumberOfColumns() == nCols);
             DAAL_ASSERT(buffer.size() == nRows * nCols);
```

Comment:
```suggestion
            DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(size_t, nRows , nCols);
            DAAL_ASSERT(buffer.size() == nRows * nCols);
```

Reply:
I'm not addressing SDL in data management now